rpc/js/package.json

```diff
@@ -23,18 +23,22 @@
     "ts-protoc-gen": "^0.14.0",
     "tslib": "^2.4.1",
     "typescript": "^5.2.2",
-    "vite": "^3.1.4",
-    "webpack": "^5.74.0",
-    "webpack-cli": "^4.10.0"
+    "vite": "^5.0.6"
+  },
+  "type": "module",
+  "types": "./dist/main.d.ts",
+  "exports": {
+    ".": {
+      "types": "./dist/main.d.ts",
+      "import": "./dist/main.js"
+    }
   },
-  "main": "dist/main.mjs",
-  "types": "src/index.d.ts",
   "scripts": {
     "prepare": "./etc/prepare.sh",
     "typecheck": "tsc --noEmit",
-    "lint": "eslint '.*.cjs' '**/*.{ts,js,cjs}' --quiet",
+    "lint": "eslint '**/*.{ts,js,cjs}' --quiet",
     "format": "prettier --write '**/*.{ts,js,cjs}'",
-    "build": "npx webpack",
+    "build": "vite build && tsc --emitDeclarationOnly",
```

Comment:
Looks like `--emitDeclarationOnly` is redundant with the `tsconfig.json` setting